ideascube/mixins.py

```diff
@@ -11,6 +11,36 @@
 from ideascube.search.models import Search
 
 
+class OrderableViewMixin:
+
+    ORDERS = []
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.ORDERS:
+            context['orders'] = self.ORDERS
+            default = self.ORDERS[0]
+            context['order_by'] = (self.request.GET.get('order_by') or
+                                   default['key'])
+            context['sort'] = self.request.GET.get('sort') or default['sort']
+        return context
+
+    def get_ordering(self):
+        key = self.request.GET.get('order_by')
+        orders = OrderedDict((o['key'], o) for o in self.ORDERS)
```

Comment:
You don't need an OrderedDict here, a simple dict would be just enough:

```python
        orders = {o['key']: o for o in self.ORDERS}
```

But going further, you don't really need the dict either:

```python
    def get_ordering(self):
        key = self.request.GET.get('order_by')

        for o in self.ORDERS:
            if o['key'] == key:
                order = o
                break
        elif self.ORDERS:
            order = self.ORDERS[0]
        else:
            return None

        order_by = order['expression']
        ...
```

I'd personally also get rid of the case where `self.ORDERS` evaluates to False as early as possible, right at the beginning of this function, because there's no point even looking at the `order_by` passed as `GET` if we can't order by anything anyway:

```python
    def get_ordering(self):
        if not self.ORDERS:
            return

        key = self.request.GET.get('order_by')

        for o in self.ORDERS:
            if o['key'] == key:
                order = o
                break
        else:
            order = self.ORDERS[0]

        order_by = order['expression']
        ...
```

But that's mere personal preference, not something critical.